client/transaction-pool/tests/pool.rs

```diff
@@ -530,8 +538,9 @@ fn fork_aware_finalization() {
 			block_on(pool.submit_and_watch(&BlockId::number(1), SOURCE, from_charlie.clone()))
 				.expect("1.Imported");
 		assert_eq!(pool.status().ready, 1);
-		let header = pool.api().push_block(3, vec![from_charlie.clone()], true);
-
+		let header = pool.api().push_block_with_parent(b1, vec![from_charlie.clone()], true);
```

Comment:
The test with the previous chain structure was not failing.

Some of events did not include the actual tree_route which would cause some of transactions to be retracted.
Lack of tree_route did not cause the retraction for D1 and E1 events. The flow of notifications for transactions was fine.


To be more precise, the test was:

```
        C2 - D2- E1
      /   \
     /     D1
A -B1
     \
      C1
```

And the sequence of events;
```
nbb(B1) []
f(B1)   []

nbb(C2) []
nbb(D2) []

nbb(C1) [ D2-C2-B1 p:2 ]
f(C1)   []

nbb(D1) []
f(D1)   []

nbb(E1) []
f(E1)   []
```

The only retraction expected by test was performed in `nbb(C1)` event, which provided the tree_route which is exactly the same as in case of _two-fork_ chain.